dispatcher_plugin_nb2workflow/dataserver_dispatcher.py

```diff
@@ -177,14 +177,50 @@ def run_query(self,
             task=self.task     
 
         if param_dict is None:
-            param_dict=self.param_dict   
-        
+            param_dict=self.param_dict
+
         if run_asynch:
             param_dict['_async_request_callback'] = call_back_url
             param_dict['_async_request'] = "yes"
 
+        spl_cb_url = urlsplit(call_back_url)
```

Comment:
this function is very long already. Could you extract this functionality out of it?

Reply:
the whole `run_query` is now more compact